soc/xtensa/intel_adsp/common/include/soc.h

```diff
@@ -70,6 +70,13 @@
 #define __imr __in_section_unique(imr)
 #define __imrdata __in_section_unique(imrdata)
 
+extern char _text_start[];
+extern char _text_end[];
+extern char _end[];
+extern char _heap_sentry[];
+extern char _cached_start[];
+extern char _cached_end[];
```

Comment:
> These create a build type errors with current SOF mainline.

@edersondisouza running SOF may require a bit of knowledge and practice but merely _building_ all supported configurations/overlays should be very easy: just try `sof/scripts/xtensa-build-zephyr.py -a / --help` and that should be it. Any problem or question with this script please contact me directly 24/7.

It is also possible to submit Zephyr PRs to SOF CI (with tests running on hardware) thanks to one-line hack in the same script.

And there is of course twister too https://github.com/zephyrproject-rtos/zephyr/pull/44529/checks?check_run_id=5826346658 but its configuration may differ a bit - different goals.

Let's make sure SOF is never "surprised" by any Zephyr incompatibility again :-)

For any further CI discussion https://github.com/thesofproject/sof/issues/3517 is a decent starting point.

Reply:
@kv2019i Why do you think changing the type for the linker declarations in the SOF side would break bisect? It appears to be a self contained code, using `char []` should just work (TM), no? I mean, I'm not changing Zephyr code, I'm _adding_ it. Using a newer SOF with `char []` should work with old Zephyr, without this code.

 OTOH, changing here for `uint8_t` conflicts with other Zephyr declarations... =/

Reply:
I created https://github.com/thesofproject/sof/pull/5648 - hopefully it can solve this issue.

Reply:
Ack @edersondisouza , change on SOF side seems best. This still breaks the bisect case if you have a workarea with newer Zephyr (with this PR merged) and you want to bisect on SOF side to debug something. Trying to build an older SOF git tree (without your SOF PR 5648) will not build. This is by no means the only such case. I've tried to use ifdefs when possible (like here https://github.com/thesofproject/sof/pull/5455/commits/e2f3eaa47f455c003fb60c784006fddb28cfd486 ), but this approach really only works if you fix Zephyr to a known stable release while bisecting. 

No need to block this PR for this, but I'm trying to find a way to get to more stable standing where we can keep bisect working with less effort.